File: B. Medium/3016-minimum-number-of-pushes-to-type-word-ii/3016-minimum-number-of-pushes-to-type-word-ii.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        #from Collections import Counter
        key_list_count = [0, 8, 8, 8, 3]
        count_word_list = sorted([(v, k) for k, v in Counter(word).items()], reverse=True)
        cost = 0
        i = 1
        for v, k in count_word_list:
            if (key_list_count[i]>0):
                cost += v * i
                key_list_count[i] -= 1
            if (key_list_count[i]==0):
                i += 1
        return cost
        

File: Daily Questions/3016-minimum-number-of-pushes-to-type-word-ii/3016-minimum-number-of-pushes-to-type-word-ii.c
// Comparison function
int compare(const void* a, const void* b) {
   return (*(int*)b - *(int*)a);
}

int minimumPushes(char* word) {
    int key_list_count[5] = {0, 8, 8, 8, 3};
    int freq[27] = {0};
    for(int i = 0; word[i] != '\0'; i++) {
        freq[(unsigned int)word[i] - 'a']++;
    }
    qsort(freq, 27, sizeof(int), compare);
    int cost = 0;
    for(int i = 0; i <= 26; i++){
        cost += freq[i] * (1 + (i / 8));
    }
    return cost;
}